File: CSE-206--AlgorithmsLab/Lab-01__2021-02-08__graph/Task__02__ArrayList.java
import java.util.Scanner;
import java.util.ArrayList;
public class Task__02__ArrayList {
	public static void main(String[] args) {
		new Graph();
	}
}
class Graph {
	Scanner input = new Scanner(System.in);
	ArrayList<ArrayList<Integer>> graphList = new ArrayList<ArrayList<Integer>>();
	int vertices;
	Graph() {
		System.out.printf("Vertices: ");
        this.vertices = input.nextInt();
		graphInput();
		graphPrint();
	}
	void graphInput() {
		int temp;
		for (int i=0; i<vertices; i++) {
			ArrayList<Integer> tempGraph = new ArrayList<Integer>();
			boolean isDone=false;
			while (!isDone) {
				temp = input.nextInt();
				if (temp == -1) isDone=true;
				else tempGraph.add(temp);
			}
			graphList.add(tempGraph);
		}
	}
	void graphPrint() {
		System.out.println("AdjacencyList: ");
		for (int i=0; i<vertices; i++) {
			int listSize = graphList.get(i).size();
			for (int j=0; j<listSize; j++) {
				System.out.print(graphList.get(i).get(j));
				if (j+1 == listSize) System.out.println();
                else System.out.printf(" ");
			}
		}
	}
}

File: CSE-206--AlgorithmsLab/Lab-01__2021-02-08__graph/Python/Task__02__ArrayList.py
vertices = int(input("Vertices: "))
graphList = []
for r in range(vertices):
    tempList = []
    isDone = False
    while not isDone:
        temp = int(input())
        if temp == -1:
            isDone = True
        else:
            tempList.append(temp)
    graphList.append(tempList)
print("AdjacencyList: ")
for r in range(vertices):
    tempLength = len(graphList[r])
    for c in range(tempLength):
        if c+1 != tempLength:
            print(graphList[r][c], end = " ")
        else:
            print(graphList[r][c])

